tests/integration/targets/aci_epg_subnet/tasks/main.yml

```diff
@@ -0,0 +1,283 @@
+# Test code for the ACI modules
+# Copyright: (c) 2022, Gaspard Micol (@gmicol)
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+- name: Set vars
+  set_fact:
+    aci_info: &aci_info
+      host: "{{ aci_hostname }}"
+      username: "{{ aci_username }}"
+      password: "{{ aci_password }}"
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: debug
+
+- name: Query system information
+  aci_system:
+    <<: *aci_info
+    id: 1
+    state: query
+  register: version
+
+- name: Verify Cloud and Non-Cloud Sites in use.
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: ensure tenant does not exists
+  cisco.aci.aci_tenant: &aci_tenant_absent
+    <<: *aci_info
+    state: absent
+    tenant: ansible_test
+
+- name: Execute tasks only for non-cloud sites
+  when: query_cloud.current == []  # This condition will execute only non-cloud sites
+  block:  # block specifies execution of tasks within, based on conditions
+  - name: ensure tenant exists for tests to kick off
+    cisco.aci.aci_tenant: &aci_tenant_present
+      <<: *aci_tenant_absent
+      state: present
+      tenant: ansible_test
+    register: tenant_present
+
+  - name: ensure ap exists for tests to kick off
+    cisco.aci.aci_ap: &aci_ap_present
+      <<: *aci_tenant_present
+      ap: anstest
+    register: ap_present
+
+  - name: ensure epg exists for tests to kick off
+    cisco.aci.aci_epg: &aci_epg_present
+      <<: *aci_ap_present
+      epg: anstest
+    register: epg_present
+
+  - name: create subnet - check mode works
+    cisco.aci.aci_epg_subnet: &aci_subnet_present
+      <<: *aci_epg_present
+      state: present
+      subnet_name: anstest
+      gateway: 10.100.100.1
+      mask: 24
+      descr: Ansible Test
+    check_mode: true
+    register: create_check_mode
+
+  - name: create subnet - creation works
+    cisco.aci.aci_epg_subnet:
+      <<: *aci_subnet_present
+    register: create_subnet
+
+  - name: create new subnet - creation works
```

Comment:
Try to maintain a unique task name if required you can add the APIC version. Please do it for all tasks.